packages/core/src/codewhispererChat/controllers/chat/messenger/messenger.ts

```diff
@@ -88,6 +88,16 @@ export class Messenger {
             )
         )
     }
+
+    public countTotalNumberOfCodeBlocks(message: string): number {
```

Comment:
We've discussed several times that this method will not work properly especially in case there are code block markdowns inside code blocks. Please use `mynahUI.endMessageStream` instead.

Reply:
Send message to webview from IDE, grab the latest count by ending the stream on presentation layer with `mynahUI.endStream` and send it back to IDE level from webview.

Reply:
I tried this approach and not able to get the `noOfCountBlocks` from mynahUi and even with this we can not send this param to the Telemetry event `recordAddTelemetry` event as this will be sent by then. So the margin of error is same for both Claude V2 and V3. Discussed with Chandan and Vladimir regarding this and went with this approach.

Reply:
As a temporary solution (until backend response contains the number of code blocks with the last chunk) i'm approving it. But definitely needs to be updated asap.